contrib/pg_tde/t/pgtde.pm

```diff
@@ -81,4 +81,65 @@ sub compare_results
 	return compare($expected_filename_with_path, $out_filename_with_path);
 }
 
+# Common TDE helpers
+
+# Check if the encryption status of a table is as expected and return 't' or 'f'
+sub check_encryption_status
+{
+	my ($node, $table_name, $expected) = @_;
+	my $result =
+	  safe_psql('postgres', "SELECT pg_tde_is_encrypted('$table_name')");
+	append_to_result_file($node->name . ": encryption check result for $table_name = $result");
+	is($result, $expected, "Check encryption status for '$table_name' on " . $node->name);
+}
+
+# Set up pg_tde extension and add a global key provider and set the server key
+sub setup_pg_tde_global_environment
+{
+	my ($node, $key_name, $provider_name, $provider_path) = @_;
+	psql($node, 'postgres', 'CREATE EXTENSION IF NOT EXISTS pg_tde;');
+	psql($node, 'postgres',
+		"SELECT pg_tde_add_global_key_provider_file('$provider_name', '$provider_path');");
+	psql($node, 'postgres',
+		"SELECT pg_tde_set_server_key_using_global_key_provider('$key_name', '$provider_name');");
+}
+
+# Set up pg_tde extension and add a database key provider and set the database key
+sub setup_pg_tde_db_environment
+{
+	my ($node, $key_name, $provider_name, $provider_path) = @_;
```

Comment:
In most cases in tests nobody cares about the key and provider name, and path. So maybe have defaults like:
```
my $testname = basename($0);
$testname =~ s/\.[^.]+$//;

my $key_name = $key_name . '_db_key';  
my $provider_name = $testname . '_provider';
my $provider_path = '/tmp/' . $testname . '.per';
```

And then in most cases you call much more convenient  `setup_pg_tde_db_key($node)`